### PR TITLE
[Kotlin Cleanup] Cardbrowser pt12

### PR DESCRIPTION
**Pull request by dorrin-sot**
Continued from #11468

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -1497,21 +1497,18 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
...
     private fun searchCards() {
         // cancel the previous search & render tasks if still running
         invalidate()
-        if (mSearchTerms == null) {
-            mSearchTerms = ""
-        }
-        if ("" != mSearchTerms && mSearchView != null) {
+        mSearchTerms = mSearchTerms ?: ""
+        if (mSearchTerms!!.isNotEmpty() && mSearchView != null) {
             mSearchView!!.setQuery(mSearchTerms!!, false)
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1505]: You can even do 
```
mSearchView?.apply {
  setQuery(mSearchTerms!!, false)
  expandActionView()
}
```

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 2]
```diff
@@ -1587,11 +1587,8 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
             .stream()
             .map { cardId: Long? -> col.getCard(cardId!!).note() }
             .distinct()
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1588]: You can even remove the value selectedNotes and directly do `.forEach`

**Review by Arthur-Milchior — COMMENTED** [review]
I really love the work you're doing. 
I'd love to know: 
* are you on discord? If so please ping me "Arthur".
* do you intend to work on card browser only, or to touch other part of the code?

in case you don't know, some part of the card browser will be rewritten by a Google Summer of Code student. I assume it'll help the student, but it also mean some part of it will be lost soon.

If you are interested in touching other part of the code too, you may want to start now. There is already so many PRs related to the browser, you're taking conflict. On the other hand, cleaning other files would ensure that those Browser PR can be considered without interaction

**Comment by dorrin-sot** [comment]
> I really love the work you're doing. I'd love to know:
> 
> * are you on discord? If so please ping me "Arthur".

Thanks! I do I'll message you later 👀

> * do you intend to work on card browser only, or to touch other part of the code?

I'm actually enjoying working with kotlin and doing cleanup alot so I'll probably try to find other big activities after, especially when I don't have any issues to work on

**Pull request merged**